sys/dev/sound/pci/hda/pin_patch_realtek.h

```diff
@@ -0,0 +1,813 @@
+	}, { }
+};
+
+static struct pin_patch_t pin_patches_dell_xps_jack[] = {
+	PIN_PATCH_JACK_WO_DETECT(24),
+	PIN_PATCH_HPMIC_WO_DETECT(26),
+	{ }
+};
+
+/*
+ * List of models and patches
+ */
+static struct hdaa_model_pin_patch_t realtek_model_pin_patches[] = {
+	{ /**** CODEC: HDA_CODEC_ALC260 ****/
+		.id = HDA_CODEC_ALC260,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(SONY_S5_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_PATCH_STRING(16, "seq=15 device=Headphones"),
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(HP_DC5750_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x11, 0x90130110), /* speaker */
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(SONY_VAIO_TX_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x0f, 0x01214000), /* hp */
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(SONY_81BBID_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x0f, 0x01211020),
+					PIN_OVERRIDE(0x10, 0x0001003f),
+					PIN_PATCH_NOT_APPLICABLE(0x11),
+					PIN_OVERRIDE(0x12, 0x01a15930),
+					PIN_PATCH_NOT_APPLICABLE(0x13),
+					PIN_PATCH_NOT_APPLICABLE(0x14),
+					PIN_PATCH_NOT_APPLICABLE(0x15),
+					PIN_PATCH_NOT_APPLICABLE(0x16),
+					PIN_PATCH_NOT_APPLICABLE(0x17),
+					PIN_PATCH_NOT_APPLICABLE(0x18),
+					PIN_PATCH_NOT_APPLICABLE(0x19),
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC268 ****/
+		.id = HDA_CODEC_ALC268,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(ACER_T5320_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_PATCH_STRING(20, "as=1 seq=15"),
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(ACER_TM_6293_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x1e, 0x014b1180),
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC269 ****/
+		.id = HDA_CODEC_ALC269,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(LENOVO_X1CRBN_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_PATCH_STRING(21, "as=1 seq=15"),
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(LENOVO_T430_SUBVENDOR),
+					PIN_SUBVENDOR(LENOVO_T430S_SUBVENDOR),
+					PIN_SUBVENDOR(LENOVO_X230_SUBVENDOR),
+					PIN_SUBVENDOR(LENOVO_X230T_SUBVENDOR),
+					PIN_SUBVENDOR(LENOVO_T431S_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_PATCH_DOCK_MIC_IN(0x19),
```

Comment:
This looks weird, when mixing hexadecimal and decimal numbers to pin number - over the whole file.

Reply:
Decimal numbers are used when the original files used decimal, hex when the original used hex. Until now, the FreeBSD originals used only decimal, the Linux sources used onnly hex. While porting, I would like to use no more changes than necessary to make debugging as easy as possible.